src/chart/components/x_axis/x-axis-labels.generator.ts

```diff
@@ -146,7 +146,7 @@ export class XAxisTimeLabelsGenerator implements XAxisLabelsGenerator {
 			),
 		);
 
-		return [...visualCandles, ...appendFakeCandle];
+		return [...prependedCandles, ...visualCandles, ...appendFakeCandle];
```

Comment:
why don't just do something like `appendFakeCandle` but with `prepend?`

Reply:
It would be incorrect name, because not always candles from the past are fake, for example, we can scroll a lot to the past (to the first candle and further) and after that change to instrument which has data for that period